plugins/repository-azure/src/internalClusterTest/java/org/elasticsearch/repositories/azure/AzureStorageCleanupThirdPartyTests.java

```diff
@@ -111,4 +117,19 @@ private void ensureSasTokenPermissions() {
         }));
         future.actionGet();
     }
+
+    public void testMultiBlockUpload() throws Exception {
+        final BlobStoreRepository repo = getRepository();
+        // The configured threshold for this test suite is 1mb
+        final int blobSize = ByteSizeUnit.MB.toIntBytes(2);
+        final Executor executor = repo.threadPool().generic();
+        PlainActionFuture<Void> future = PlainActionFuture.newFuture();
+        executor.execute(ActionRunnable.run(future, () -> {
```

Comment:
NIT: just inline `repo.threadPool().generic()` here, you're only using `executor` once